zwave_js_server/model/controller/__init__.py

```diff
@@ -830,6 +646,10 @@ def receive_event(self, event: Event) -> None:
                 f"Controller doesn't know how to handle/forward this event: {event.data}"
             )
 
+        validate_event_data(
```

Comment:
Shouldn't we return the pydantic model instance here and use it? That's one of the big advantages afterwards that we have a typed higher level instance to use instead of a dict with base types. Is that the second step of the refactor?

Reply:
Looks like we can use this function to effectively create a new model: https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L610

Reply:
That method doesn't validate the update data, and this issue has some more ideas on how to solve for it: https://github.com/samuelcolvin/pydantic/issues/1673